src/pydrex/minerals.py

```diff
@@ -98,185 +100,155 @@ class Mineral:
-        fractions_iter.clip(0, None)
-        fractions_iter /= fractions_iter.sum()
-
-        # ========== RK step 3 ==========
-        orientations_diff, fractions_diff = _core.derivatives(
-            phase=self.phase,
-            fabric=self.fabric,
-            n_grains=self.n_grains,
-            orientations=self.orientations,
-            fractions=self.fractions,
-            strain_rate=strain_rate,
-            velocity_gradient=velocity_gradient,
-            stress_exponent=stress_exponent,
-            dislocation_exponent=dislocation_exponent,
-            nucleation_efficiency=nucleation_efficiency,
-            gbm_mobility=gbm_mobility,
-            volume_fraction=volume_fraction,
-        )
-        orientations_3 = orientations_diff * dt * strain_rate_max
-        orientations_iter = self.orientations + orientations_3
-        orientations_iter.clip(-1, 1)
-        fractions_3 = fractions_diff * dt * strain_rate_max
-        fractions_iter = self.fractions + fractions_3
-        fractions_iter.clip(0, None)
-        fractions_iter /= fractions_iter.sum()
-
-        # ========== RK step 4 ==========
-        orientations_diff, fractions_diff = _core.derivatives(
-            phase=self.phase,
-            fabric=self.fabric,
-            n_grains=self.n_grains,
-            orientations=self.orientations,
-            fractions=self.fractions,
-            strain_rate=strain_rate,
-            velocity_gradient=velocity_gradient,
-            stress_exponent=stress_exponent,
-            dislocation_exponent=dislocation_exponent,
-            nucleation_efficiency=nucleation_efficiency,
-            gbm_mobility=gbm_mobility,
-            volume_fraction=volume_fraction,
-        )
-        orientations_4 = orientations_diff * dt * strain_rate_max
-        self.orientations = (
-            self.orientations
-            + (
-                orientations_1 / 2
-                + orientations_2
-                + orientations_3
-                + orientations_4 / 2
-            )
-            / 3
-        )
-        self.orientations.clip(-1, 1)
-        fractions_4 = fractions_diff * dt * strain_rate_max
-        self.fractions += (
-            fractions_1 / 2 + fractions_2 + fractions_3 + fractions_4 / 2
-        ) / 3
-        self.fractions /= self.fractions.sum()
-
-        # Grain boundary sliding for small grains.
-        mask = self.fractions < gbs_threshold / self.n_grains
-        self.orientations[mask, :, :] = self._orientations_init[mask, :, :]
-        self.fractions[mask] = gbs_threshold / self.n_grains
-        self.fractions /= self.fractions.sum()
+
+        while sol.status == "running":
+            sol.step()
+            finite_strain_ell, orientations, fractions = extract_vars(sol.y)
+            # Grain boundary sliding for small grains.
+            mask = fractions < config["gbs_threshold"] / self.n_grains
+            # No rotation: carry over previous orientations.
```

Comment:
Can you actually check if there is some mention of this somewhere in the DRex papers? Carrying over previous orientations sounds fair, but the Fortran reverts to the initial value, as we discussed.